sdk/include/platform/sunburst/platform-gpio.hh

```diff
@@ -313,62 +338,140 @@ struct SonataGpioRaspberryPiHat : SonataGpioBase<SonataGpioRaspberryPiHat>
  * Documentation source:
  * https://lowrisc.org/sonata-system/doc/ip/gpio.html
  */
-struct SonataGpioArduinoShield : SonataGpioBase<SonataGpioArduinoShield>
+struct SonataGpioArduinoShield : SonataGpioBase
 {
 	/**
 	 * The mask of bits of the `output` register that contain meaningful GPIO
 	 * output.
 	 */
-	static const uint32_t OutputMask = 0x0003'FFFF;
+	static constexpr uint32_t OutputMask()
+	{
+		return 0x0000'3FFF;
+	}
 
 	/**
 	 * The mask of bits of the `input` and `debouncedInput` registers that
 	 * contain meaningful GPIO input.
 	 */
-	static const uint32_t InputMask = 0x0003'FFFF;
+	static constexpr uint32_t InputMask()
+	{
+		return 0x0000'3FFF;
+	}
 
 	/**
 	 * The mask of bits of the `outputEnable` register that correspond to GPIO
 	 * pins which can have their output enabled or disabled.
 	 */
-	static const uint32_t OutputEnableMask = 0x0003'FFFF;
+	static constexpr uint32_t OutputEnableMask()
+	{
+		return 0x0000'3FFF;
+	}
 };
 
 /**
- * A driver for Sonata's PMOD Header GPIO.
+ * A driver for Sonata's PMOD0 Header GPIO.
  *
  * Documentation source:
  * https://lowrisc.org/sonata-system/doc/ip/gpio.html
  */
-struct SonataGpioPmod : SonataGpioBase<SonataGpioPmod>
+struct SonataGpioPmod0 : SonataGpioBase
 {
 	/**
 	 * The mask of bits of the `output` register that contain meaningful GPIO
 	 * output.
 	 */
-	static const uint32_t OutputMask = 0x0000'FFFF;
+	static constexpr uint32_t OutputMask()
+	{
+		return 0x0000'00FF;
+	}
 
 	/**
 	 * The mask of bits of the `input` and `debouncedInput` registers that
 	 * contain meaningful GPIO input.
 	 */
-	static const uint32_t InputMask = 0x0000'FFFF;
+	static constexpr uint32_t InputMask()
+	{
+		return 0x0000'00FF;
+	}
 
 	/**
 	 * The mask of bits of the `outputEnable` register that correspond to GPIO
 	 * pins which can have their output enabled or disabled.
 	 */
-	static const uint32_t OutputEnableMask = 0x0000'FFFF;
+	static constexpr uint32_t OutputEnableMask()
+	{
+		return 0x0000'00FF;
+	}
+};
 
+/**
+ * A driver for Sonata's PMOD1 Header GPIO.
+ *
+ * Documentation source:
+ * https://lowrisc.org/sonata-system/doc/ip/gpio.html
+ */
+struct SonataGpioPmod1 : SonataGpioBase
```

Comment:
One `SonataGpioPmod` which is aliased to `SonataGpioPmod{0,1}`

Reply:
Latest force push makes this change.